mempool/mempool.go

```diff
@@ -524,20 +523,26 @@ func (cache *mapTxCache) Push(tx types.Tx) bool {
 	if cache.list.Len() >= cache.size {
 		popped := cache.list.Front()
 		poppedTx := popped.Value.(types.Tx)
-		// NOTE: the tx may have already been removed from the map
-		// but deleting a non-existent element is fine
```

Comment:
so we knew we weren't removing from the list in .Remove but we thought it was ok?

Reply:
I'm guessing noone considered the lingering memory or the issues with how it affects cache eviction of good txs. (Full linked list, empty hashmap)